sklearn/preprocessing/tests/test_data.py

```diff
@@ -1939,11 +1937,13 @@ def test_normalizer_l1():
     for X in (X_dense, X_sparse_pruned, X_sparse_unpruned):
 
         normalizer = Normalizer(norm='l1', copy=True)
+        normalizer.n_features_in_ = X.shape[1]
```

Comment:
This means that `Normalizer` used to be stateless but it's no longer the case?

Shouldn't we make it so that when `n_features_in_` is missing we just skip the check in `_validate_data` when `reset=False` (maybe at least for estimators with the `"stateless"` estimator tag)?